File: HelpBot/src/main/java/com/example/HelpBot/log/IHBLogger.java
package com.example.HelpBot.log;

public interface IHBLogger
{
    void d(final String tag, final String message);

    void w(final String tag, final String message);

    void e(final String tag, final String message);

    void d(final String tag, final String message, final Throwable tr);

    void w(final String tag, final String message, final Throwable tr);

    void e(final String tag, final String message, final Throwable tr);

    public enum LEVEL
    {
        DEBUG,
        WARN,
        ERROR;
    }
}

File: SDK/iOS/HelpBotSDK/Sources/HelpBotSDK/Log/IHBLogger.swift
import Foundation

/**
 HelpBot 统一日志协议（由宿主注入）。

 - 注意：SDK 作为依赖库，不应强行接管宿主日志系统。
 - 安全：HBlogger 会对敏感字段做脱敏处理，且默认仅在 DEBUG 构建允许输出。
 */
public protocol IHBLogger: AnyObject {
    func d(_ tag: String, _ message: String, _ error: Error?)
    func w(_ tag: String, _ message: String, _ error: Error?)
    func e(_ tag: String, _ message: String, _ error: Error?)
}


